File: CS/CustomTileControl.cs
using DevExpress.Data.Filtering;
using DevExpress.Utils;
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace dxExample {
    public class CustomTileControl : TileControl, ISearchControlClient {
        public CustomTileControl() {

        }
        ISearchControl srchControl;
        public void ApplyFindFilter(SearchInfoBase searchInfo) {
            if (searchInfo == null) {
                foreach (TileGroup group in this.Groups) {
                    foreach (TileItem item in group.Items)
                        item.Visible = true;
                }
            }
            else {
                TileSearchInfo info = searchInfo as TileSearchInfo;
                switch (info.FilterCondition) {
                    case FilterCondition.StartsWith:
                        foreach (TileGroup group in this.Groups) {
                            foreach (TileItem item in group.Items)
                                item.Visible = item.Text.ToUpper().StartsWith(info.SearchText.ToUpper());
                        }
                        break;
                    case FilterCondition.Contains:
                    case FilterCondition.Like:
                    case FilterCondition.Default:
                        foreach (TileGroup group in this.Groups) {
                            foreach (TileItem item in group.Items)
                                item.Visible = item.Text.ToUpper().Contains(info.SearchText.ToUpper());
                        }
                        break;
                    case FilterCondition.Equals:
                        foreach (TileGroup group in this.Groups) {
                            foreach (TileItem item in group.Items)
                                item.Visible = item.Text.ToUpper().Equals(info.SearchText.ToUpper());
                        }
                        break;
                }
            }
        }

        public SearchControlProviderBase CreateSearchProvider() {
            return new TileSearchProvider();
        }

        public bool IsAttachedToSearchControl {
            get { return srchControl != null; }
        }

        public void SetSearchControl(ISearchControl searchControl) {
            if (srchControl == searchControl) return;
            srchControl = searchControl;
            ApplyFindFilter(null);
        }
    }
}

File: VB/CustomTileControl.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Data.Filtering
Imports DevExpress.Utils
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace dxExample
	Public Class CustomTileControl
		Inherits TileControl
		Implements ISearchControlClient
		Public Sub New()

		End Sub
		Private srchControl As ISearchControl
        Public Sub ApplyFindFilter(ByVal searchInfo As SearchInfoBase) Implements ISearchControlClient.ApplyFindFilter
            If searchInfo Is Nothing Then
                For Each group As TileGroup In Me.Groups
                    For Each item As TileItem In group.Items
                        item.Visible = True
                    Next item
                Next group
            Else
                Dim info As TileSearchInfo = TryCast(searchInfo, TileSearchInfo)
                Select Case info.FilterCondition
                    Case FilterCondition.StartsWith
                        For Each group As TileGroup In Me.Groups
                            For Each item As TileItem In group.Items
                                item.Visible = item.Text.ToUpper().StartsWith(info.SearchText.ToUpper())
                            Next item
                        Next group
                    Case FilterCondition.Contains, FilterCondition.Like, FilterCondition.Default
                        For Each group As TileGroup In Me.Groups
                            For Each item As TileItem In group.Items
                                item.Visible = item.Text.ToUpper().Contains(info.SearchText.ToUpper())
                            Next item
                        Next group
                    Case FilterCondition.Equals
                        For Each group As TileGroup In Me.Groups
                            For Each item As TileItem In group.Items
                                item.Visible = item.Text.ToUpper().Equals(info.SearchText.ToUpper())
                            Next item
                        Next group
                End Select
            End If
        End Sub

        Public Function CreateSearchProvider() As SearchControlProviderBase Implements ISearchControlClient.CreateSearchProvider
            Return New TileSearchProvider()
        End Function

        Public ReadOnly Property IsAttachedToSearchControl() As Boolean Implements ISearchControlClient.IsAttachedToSearchControl
            Get
                Return srchControl IsNot Nothing
            End Get
        End Property

        Public Sub SetSearchControl(ByVal searchControl As ISearchControl) Implements ISearchControlClient.SetSearchControl
            If srchControl Is searchControl Then
                Return
            End If
            srchControl = searchControl
            ApplyFindFilter(Nothing)
        End Sub
	End Class
End Namespace
